PWGJE/Tasks/trackEfficiency.cxx

```diff
@@ -0,0 +1,235 @@
+
+    registry.add("h3_track_pt_track_eta_track_phi_nonassociatedtrack", "#it{p}_{T, track} (GeV/#it{c}); #eta_{track}; #phi_{track}", {HistType::kTH3F, {{500, 0., 10.}, {100, -1.0, 1.0}, {400, -1.0, 7.}}});
+    registry.add("h3_track_pt_track_eta_track_phi_associatedtrack_primary", "#it{p}_{T, track} (GeV/#it{c}); #eta_{track}; #phi_{track}", {HistType::kTH3F, {{500, 0., 10.}, {100, -1.0, 1.0}, {400, -1.0, 7.}}});
+    registry.add("h3_track_pt_track_eta_track_phi_associatedtrack_nonprimary", "#it{p}_{T, track} (GeV/#it{c}); #eta_{track}; #phi_{track}", {HistType::kTH3F, {{500, 0., 10.}, {100, -1.0, 1.0}, {400, -1.0, 7.}}});
+    registry.add("h3_track_pt_track_eta_track_phi_associatedtrack_split", "#it{p}_{T, track} (GeV/#it{c}); #eta_{track}; #phi_{track}", {HistType::kTH3F, {{500, 0., 10.}, {100, -1.0, 1.0}, {400, -1.0, 7.}}});
+
+    registry.add("h3_particle_pt_particle_eta_particle_phi_associatedtrack_primary", "#it{p}_{T, mcpart} (GeV/#it{c}); #eta_{mcpart}; #phi_{mcpart}", {HistType::kTH3F, {{500, 0., 10.}, {100, -1.0, 1.0}, {400, -1.0, 7.}}});
+    registry.add("h3_particle_pt_particle_eta_particle_phi_associatedtrack_nonprimary", "#it{p}_{T, mcpart} (GeV/#it{c}); #eta_{mcpart}; #phi_{mcpart}", {HistType::kTH3F, {{500, 0., 10.}, {100, -1.0, 1.0}, {400, -1.0, 7.}}});
+    registry.add("h3_particle_pt_particle_eta_particle_phi_associatedtrack_split", "#it{p}_{T, mcpart} (GeV/#it{c}); #eta_{mcpart}; #phi_{mcpart}", {HistType::kTH3F, {{500, 0., 10.}, {100, -1.0, 1.0}, {400, -1.0, 7.}}});
+  }
+
+  Preslice<JetTracksMCD> tracksPerJCollision = o2::aod::jtrack::collisionId;
+
+  void process(JetMcCollision const& mcCollision,
+               soa::SmallGroups<JetCollisionsMCD> const& collisions, // smallgroups gives only the collisions associated to the current mccollision, thanks to the mccollisionlabel pre-integrated in jetcollisionsmcd
+               soa::Join<JetTracksMCD, aod::JTrackExtras> const& jetTracks,
+               JetParticlesWithOriginal const& jMcParticles)
+  {
+    // missing:
+    //   * constexpr auto hasCentrality = CollisionMCRecTableCentFT0C::template contains<aod::CentFT0Cs>();
+    //           if constexpr (hasCentrality) {
+    //   * dividing in centrality bins
+    // I should maybe introduce the sel8 cuts on the collisoins (reco, but what about mccoll? maybe not htat way included in efficiency)
+
+    registry.fill(HIST("hMcCollCutsCounts"), 0.5); // all mcCollisions
+
+    if (!(abs(mcCollision.posZ()) < vertexZCut)) {
+      return;
+    }
+    registry.fill(HIST("hMcCollCutsCounts"), 1.5); // mcCollision.posZ() condition
+
+    if (collisions.size() < 1) {
+      return;
+    }
+    registry.fill(HIST("hMcCollCutsCounts"), 2.5); // mcCollisions with at least one reconstructed collision
+
+    if (!splitMcCollOK && collisions.size() > 1) {
```

Comment:
do you want to fully reject split collisions or also have an option to still take the first collision of the splits?